cmd/endpoint-ellipses.go

```diff
@@ -95,9 +113,14 @@ func getSetIndexes(args []string, totalSizes []uint64) (setIndexes [][]uint64, e
 		setSize = commonSize
 	}
 
-	// isValidSetSize - checks whether given count is a valid set size for erasure coding.
-	isValidSetSize := func(count uint64) bool {
-		return (count >= setSizes[0] && count <= setSizes[len(setSizes)-1] && count%2 == 0)
+	if customSetDriveCount > 0 {
+		if customSetDriveCount > setSize {
+			return nil, uiErrInvalidErasureSetSize(nil)
+		}
+		if setSize%customSetDriveCount != 0 {
```

Comment:
I am not sure what's the purpose of this.

Besides it has this problem:

```
12:56 $ MINIO_ERASURE_SET_DRIVE_COUNT=6 minio server /mnt/p4/vadmeste/xl/{1..16}
ERROR Invalid command line arguments: Invalid erasure set size.
      > Please check the passed value.
      HELP:
        Erasure set can only accept any of [4, 6, 8, 10, 12, 14, 16] values.
```

Reply:
You can only provide a custom set drive count which is divisible by the GCD for the entire set. Otherwise you cannot have uniform set distribution. 

A classic example here is 64 drives and you want to split them by 6 it's not possible to do so uniformly. The check here ensures that we do not accept such values and end up with inconsistent sets.

Reply:
Will change it .